### ISSUE TITLE
TokenManager.get promise doesn't resolve

### ISSUE THREAD

**Issue by peterszerzo** [okta/okta-auth-js#157]
In SDK version 2.0.1, I am running into cases when the following code:

```
const authClient = new OktaAuth({ /* config that usually works */ })

authClient.tokenManager.get("token")
  .then(() => console.log("then"))
  .catch(() => console.log("catch"))
```

Neither resolves nor rejects for a long time, despite the following console error showing up in the console pretty much instantaneously:
```
error.errorCode: login_required, error.description: The client specified not to prompt, but the user is not logged in.
```

A bit tricky to give more context than this, because the issue is hard to reproduce, but all in all I'm having a difficult time handling all the errors to either run a frontend application or show an error screen. I would appreciate any pointers, and happy to give more context if necessary.

**Comment by rodoabad** [comment]
@peterszerzo looks like this is where the upstream problem is starting to happen.

See https://github.com/okta/okta-oidc-js/issues/266 for more downstream issues.

**Comment by jmelberg-okta** [comment]
Hi @peterszerzo,

The method itself is throwing as it fails to renew a token. We saw this (and provided a fix) for our integration libraries.

Update your token getter by surrounding it with a `try catch`:

```javascript
const authClient = new OktaAuth({ /* config  */ });

let token;
try {
  token = await authClient.tokenManager.get("token");
} catch (e) {
  console.log(e);
  // error.errorCode: login_required,
  // error.description: The client specified not to prompt, but the user is not logged in.
}
```

Also, it is possible we're not handling a thrown error somewhere, so we'll investigate this a bit more before closing it out.

**Comment by peterszerzo** [comment]
Awesome, thanks @jmelberg-okta for the tip. This is a lot better than the wrap-in-timeout solution we found (which artificially slows down the frontend).

**Comment by aarongranick-okta** [comment]
This was fixed in version [2.9.0](https://github.com/okta/okta-auth-js/releases/tag/okta-auth-js-2.9.0)
